components/tidb_query/src/codec/row/v2/row_slice.rs

```diff
@@ -0,0 +1,261 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+#![allow(dead_code)]
+use crate::codec::{Error, Result};
+use codec::prelude::*;
+use num_traits::PrimInt;
+use tikv_util::codec::read_slice;
+
+#[derive(Default, Debug)]
+struct RowSlice<'a> {
+    is_big: bool,
+    non_null_ids: &'a [u8],
+    null_ids: &'a [u8],
+    offsets: &'a [u16],
+
+    non_null_ids_big: &'a [u32],
+    null_ids_big: &'a [u32],
+    offsets_big: &'a [u32],
+
+    values: &'a [u8],
+}
+
+impl RowSlice<'_> {
+    fn from_bytes(mut data: &[u8]) -> Result<RowSlice> {
+        assert_eq!(data.read_u8()?, super::CODEC_VERSION);
```

Comment:
For this assertion I think this function deserves a doc comment, documenting the panic scenario.

Reply:
done